### PR TITLE
EVG-13510: give more verbose and dataful host termination reason

### PR DESCRIPTION
**Pull request by hhoke**

### CODE REVIEW

File: units/host_monitoring_idle_termination.go [hunk 1]
```diff
@@ -221,10 +221,17 @@ func (j *idleHostJob) checkAndTerminateHost(ctx context.Context, h *host.Host) e
 	}
 
 	// if we haven't heard from the host or it's been idle for longer than the cutoff, we should terminate
-	if communicationTime >= idleThreshold || idleTime >= idleThreshold {
+	terminateReason := ""
+
+	if communicationTime >= idleThreshold {
```
Comment (ybrill) [inline units/host_monitoring_idle_termination.go:226]: would it be helpful to log the reason to splunk too? This reason just ends up in the db/UI, I think.
I guess it depends: why is it important to know which of these is the reason?

Comment (hhoke) [reply units/host_monitoring_idle_termination.go:226]: I had just assumed this was logged to splunk. Will make that change.

**Review by Kimchelly — APPROVED** [review]

File: units/host_termination.go [hunk 2]
```diff
@@ -344,6 +344,13 @@ func (j *hostTerminationJob) Run(ctx context.Context) {
 			"job_type": j.Type().Name,
 		}))
 		return
+	} else {
```
Comment (ybrill) [inline units/host_termination.go:347]: the else clause [is superfluous](https://golang.org/doc/effective_go.html#if): we will have returned on 346 if err != nil

File: units/host_termination.go [hunk 3]
```diff
@@ -344,6 +344,13 @@ func (j *hostTerminationJob) Run(ctx context.Context) {
...
+		grip.Info(message.Fields{
+			"message": "host successfully terminated",
```
Comment (ybrill) [inline units/host_termination.go:349]: is there any other information that would make this easier to consume in splunk? (like the distro id, for instance)

**Review by ybrill — APPROVED** [review]

**Pull request merged**